prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -263,12 +246,16 @@ class Hl7Serializer(
             // First check that we have an HL7 message we can parse.  Note some older messages may have
             // only MSH 9-1 and MSH-9-2, or even just MSH-9-1, so we need use those two fields to compare
             val msgType = PreParser.getFields(cleanedMessage, "MSH-9-1", "MSH-9-2")
+            val altMsgType = PreParser.getFields(cleanedMessage, "MSH-9-3")
```

Comment:
MSH-9-3 can also have the message ID, and it's possible MSH-9-1 and MSH-9-2 don't, so we should look at all possible combinations.